### PR TITLE
WIP: JS Governance

### PR DESCRIPTION
**Pull request by achamayou**
Putting this up for early review/help from anyone interested.

### CODE REVIEW

File: tests/governance_js.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    with primary.client(None, "member0") as c:
+        r = c.post("/gov/proposals.js", no_args_set_recovery_threshold + bad_arg_set_recovery_threshold)
+        assert r.status_code == 400 and r.body.json()["error"]["code"] == "ProposalFailedToValidate", r.body.text()
+
+    return network
+
+@reqs.description("Test proposal storage")
+def test_proposal_storage(network, args):
+    primary, _ = network.find_nodes()
+    add_member = [action("add_member", cert="", enc_pubk="", member_data={})]
```
Comment (letmaik) [inline tests/governance_js.py:45]: Should probably be wrapped in `{"actions": ...}` for future extensibility.

File: tests/governance_js.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        pid = r.body.json()["proposal_id"]
+
+        vote = {"ballot": "export function vote (proposal, proposer_id, tx) { return true }"}
+        r = c.post(f"/gov/proposals.js/{pid}/votes", vote)
+        assert r.status_code == 200, r.body.text()
+
+        member_id = 0
+        r = c.get(f"/gov/proposals.js/{pid}/votes/{member_id}")
+        assert r.status_code == 200, r.body.text()
+        assert r.body.text() == f'"{vote["ballot"]}"'
```
Comment (letmaik) [inline tests/governance_js.py:81]: I guess this is wip, should be the same schema as what's used for submission: `{"ballot": ..}`

File: src/node/rpc/member_frontend.h [hunk 3]
```diff
@@ -2058,6 +2062,371 @@ namespace ccf
...
+
+          if (!pi_)
+          {
+            return make_error(
+              HTTP_STATUS_INTERNAL_SERVER_ERROR,
+              ccf::errors::InternalError,
+              fmt::format("No proposal info associated with {} exists.", proposal_id));
+          }
+
+          // TODO: staple proposal inline here
```
Comment (letmaik) [inline src/node/rpc/member_frontend.h:2303]: Agree

File: src/node/rpc/member_frontend.h [hunk 4]
```diff
@@ -2058,6 +2062,371 @@ namespace ccf
...
+          assert(JS_VALUE_GET_TAG(module) == JS_TAG_MODULE);
+          auto module_def = (JSModuleDef*)JS_VALUE_GET_PTR(module);
+          /*
+          if (JS_GetModuleExportEntriesCount(module_def) != 1)
+          {
+            throw std::runtime_error(
+              "Endpoint module exports more than one function");
+          }
+          */
+          auto export_func = JS_GetModuleExportEntry(context, module_def, 0);
```
Comment (letmaik) [inline src/node/rpc/member_frontend.h:2168]: In your other PR you have `Context::function()`, I guess we want a similar one that accepts a function name. Should this be done in this PR or the other one?

File: tests/governance_js.py (hunk 1 above)
Comment (achamayou) [reply tests/governance_js.py:45]: Yes, good point.

File: tests/governance_js.py (hunk 2 above)
Comment (achamayou) [reply tests/governance_js.py:81]: It can be, or it can stay unwrapped, I'm happy either way.

File: src/node/rpc/member_frontend.h (hunk 4 above)
Comment (achamayou) [reply src/node/rpc/member_frontend.h:2168]: Yeah that's on the way separately.

File: tests/governance_js.py (hunk 2 above)
Comment (letmaik) [reply tests/governance_js.py:81]: I would always wrap, so that you can send more info in the future in extra fields.

**Pull request closed without merging**